streamloader/src/main/scala/com/snowplowanalytics/snowplow/storage/bigquery/streamloader/StreamLoader.scala

```diff
@@ -0,0 +1,144 @@
+import com.snowplowanalytics.snowplow.badrows.BadRow
+import com.snowplowanalytics.snowplow.storage.bigquery.common.Config.Environment
+import com.snowplowanalytics.snowplow.storage.bigquery.streamloader.Source.PubsubSource
+import com.snowplowanalytics.snowplow.storage.bigquery.streamloader.Sinks.{Bigquery, PubSub}
+import com.snowplowanalytics.snowplow.storage.bigquery.streamloader.Sinks.PubSub.PubSubOutput
+
+object StreamLoader {
+
+  //      TOPOLOGY OVERVIEW
+  //                                            +-----------+
+  //                                 bad sink   |   PubSub  |
+  //                              +-------------|   (bad)   |
+  //                              |             +-----------+
+  //                              |
+  //  +------------+          +--------+                                       aggregate
+  //  |            |  parse   | (bad,  |                  enqueue   +-------+  >> sink     +--------+
+  //  | enriched   | -------- | good)  |                  types     |types  |  types       | PubSub |
+  //  |            |          |        |               +------------|queue  | -------------| (types)|
+  //  +------------+          +--------+               |            +-------+              +--------+
+  //                              |                    |
+  //                              |                    |
+  //                              |             +------------+                              +---------+
+  //                              |             |            |              failed inserts  | PubSub  |
+  //                              +-------------| good sink  |             +----------------| (failed |
+  //                                            |            |             |                | inserts)|
+  //                                            +------|-----+             |                +---------+
+  //                                                   |                   |
+  //                                                   |             +-----|----+
+  //                                                   |             | BigQuery |
+  //                                                   +-------------| sink     |
+  //                                                                 +-----|----+
+  //                                                                       |
+  //                                                                       |                 +--------+
+  //                                                                       |     insert      |BigQuery|
+  //                                                                       +-----------------|(events)|
+  //                                                                                         +--------+
+
+  private val MaxConcurrency = Runtime.getRuntime.availableProcessors * 16
+  private val GroupByN       = 10
+  private val TimeWindow     = 5.minutes
```

Comment:
I think this value was coming from original Dataflow BQ Loader, but I don't see why this one cannot be much shorter.

Reply:
These are two separate questions.

Should it be configurable? It will add setup and maintenance burden, so if we make it configurable there needs to be a clear payoff, eg in terms of performance vs cost. But if we think that we can achieve optimal performance with negligible cost implications, then I don't see any benefit in adding one more config option.

It can be shorter, but I'm not sure I understand why it's okay here but not in the Dataflow loader. I would err on the side of consistency if there isn't a compelling reason to diverge.